ansible/README.md

```diff
@@ -196,6 +196,42 @@ ansible-playbook -i environments/$ENVIRONMENT routemgmt.yml
 - To use the API Gateway, you'll need to run `apigateway.yml` and `routemgmt.yml`.
 - Use `ansible-playbook -i environments/$ENVIRONMENT openwhisk.yml` to avoid wiping the data store. This is useful to start OpenWhisk after restarting your Operating System.
 
+### Deploying Using MongoDB
+
+You can choose MongoDB instead of CouchDB as the database backend to store entities.
+
+- Deploy a mongodb server(Optional, for test and develop only, use an external MongoDB server in production)
+
+```
+ansible-playbook -i environments/<environment> mongodb.yml -e mongodb_data_volume="/tmp/mongo-data"
+```
+
+- Then execute
+
+```
+cd <openwhisk_home>
+./gradlew distDocker
+cd ansible
+ansible-playbook -i environments/<environment> initMongodb.yml -e mongodb_connect_string="mongodb://172.17.0.1:27017"
```

Comment:
What is the relation with `db_local.ini`?

Getting this when deploying this change.

```
TASK [prepare db_local.ini] *********************************************************************************************************************************************************
Monday 24 May 2021  15:07:04 +0900 (0:00:00.883)       0:00:02.373 ************
fatal: [localhost -> localhost]: FAILED! => {"changed": false, "msg": "AnsibleUndefinedVariable: 'dict object' has no attribute 'db'"}

AnsibleUndefinedVariable: 'dict object' has no attribute 'db'

PLAY RECAP **************************************************************************************************************************************************************************
localhost                  : ok=2    changed=0    unreachable=0    failed=1
```

Reply:
Seems `pymongo` needs to be installed.
then it should be documented somewhere.

```
TASK [create necessary auth keys] ******************************************************************************************************************************************************************************************************************************************************************************************************************************************
Monday 24 May 2021  15:57:29 +0900 (0:00:01.872)       0:00:02.229 ************
failed: [ansible] (item=guest) => {"changed": false, "item": "guest", "msg": "the python pymongo module is required"}
failed: [ansible] (item=whisk.system) => {"changed": false, "item": "whisk.system", "msg": "the python pymongo module is required"}

All items completed

PLAY RECAP *****************************************************************************************************************************************************************************************************************************************************************************************************************************************************************
ansible                    : ok=1    changed=0    unreachable=0    failed=1
```

Reply:
Not quite sure the reason yet but I am unable to deploy this change with this line.

```
scheduler:
  dataManagementService:
    retryInterval: "{{ scheduler_dataManagementService_retryInterval | default(1 second) }}"
```


```
TASK [kafka : add kafka default env vars] **********************************************************************************************************************************************************************************************************************************************************************************************************************************
Monday 24 May 2021  16:07:39 +0900 (0:00:00.054)       0:00:13.372 ************
fatal: [kafka0]: FAILED! => {"msg": "An unhandled exception occurred while templating '{% set ret = [] %}{% for host in groups['zookeepers'] %}{{ ret.append( hostvars[host].ansible_host + ':' + ((zookeeper.port+loop.index-1)|string) ) }}{% endfor %}{{ ret | join(',') }}'. Error was a <class 'ansible.errors.AnsibleError'>, original message: template error while templating string: expected token ',', got 'second'. String: {{ scheduler_dataManagementService_retryInterval | default(1 second) }}"}

An unhandled exception occurred while templating '{% set ret = [] %}{% for
host in groups['zookeepers'] %}{{ ret.append( hostvars[host].ansible_host + ':'
+ ((zookeeper.port+loop.index-1)|string) ) }}{% endfor %}{{ ret | join(',') }}'.
Error was a <class 'ansible.errors.AnsibleError'>, original message: template
error while templating string: expected token ',', got 'second'. String: {{
scheduler_dataManagementService_retryInterval | default(1 second) }}

Reply:
I will remove db_local.ini relation

Reply:
AFAIK, `db_local.ini` is being used to specify the target DB, can we apply the same for MongoDB as well?

Reply:
the `db_provider` used to choose local CouchDB or ClouDant

Reply:
I didn't get error with 
```
scheduler:
  dataManagementService:
    retryInterval: "{{ scheduler_dataManagementService_retryInterval | default(1 second) }}"
```

which version of ansible do you use?

Reply:
I am using the `2.5.2` version.

Reply:
for `2.5.2` we need to add **' '** around string value, like `default('1 second')`

Reply:
oh ok, then that should be updated as we guide users to use `ansible=2.5.2`.
https://github.com/apache/openwhisk/tree/master/ansible

Reply:
ok, I will create a new PR to fix this

Reply:
We could move to a more modern version of ansible (in separate PRs of course).  We don't have to stay on 2.5.2 forever.

Reply:
> We could move to a more modern version of ansible (in separate PRs of course). We don't have to stay on 2.5.2 forever.

I'm install OW on a clean new machine and I'm updating the versions that worked.